### PR TITLE
Add disclaimer that storing secrets via k8s delegates security to customer

### PR DESCRIPTION
**Pull request by ngrayluna**
## Description

First draft of note about delegating responsibility to customers for managing secrets.

## Ticket

https://wandb.atlassian.net/browse/DOCS-795?atlOrigin=eyJpIjoiNDM0NDNjODYyOWViNGU4YTlmOWNmOGI5ODUzZWE4YWMiLCJwIjoiaiJ9

## Checklist

Check if your PR fulfills the following requirements. Put an `X` in the boxes that apply.

- [X] Files I edited were previewed on my local development server with `yarn start`. My changes did not break the local preview.
- [X] Build (`yarn docusaurus build`) was run locally and successfully without errors or warnings.
- [X] I merged the latest changes from `main` into my feature branch before submitting this PR.

### CODE REVIEW

File: docs/guides/model_registry/automation.md [hunk 1]
```diff
@@ -73,6 +68,18 @@ Specify the secrets you want to use for your webhook automation when you configu
...
+:::caution
+Secrets are available if you use either:
+* W&B SaaS public cloud or 
+* W&B Server in an Azure or GCP deployment. 
+
+Connect with your W&B account team to discuss how you can use secrets in W&B if you are on a deployment type or environment not listed above.
+
+<!-- Store secrets in a W&B instance of the cloud provider's respective secrets manager) -->
+
+You are responsible for configuring security measures that satisfy your security needs. W&B strongly recommends that you use a secrets manager provided by AWS, GCP, or Azure. Secret managers provided by AWS, GCP, and Azure are configured with advanced security capabilities.
```
Comment (noaleetz) [inline docs/guides/model_registry/automation.md]: The very last thing I would specify: a user cannot bring their own secret store (for example their own instance of Azure Key Vault). We store secrets in a **W&B** **instance** of the cloud provider's respective secrets manager. The user cannot bring their own secret store ([thread](https://weightsandbiases.slack.com/archives/C05GS5F8TNU/p1698690405841979?thread_ts=1698422606.394809&cid=C05GS5F8TNU))

Comment (noaleetz) [reply docs/guides/model_registry/automation.md]: Maybe we should throw out k8s as an option in this line: 
Before - W&B strongly recommends that you use a secrets manager provided by AWS, GCP, or Azure

After -  W&B strongly recommends that you use a secrets manager provided by AWS, GCP, or Azure, though Kubernetes is offered as a fallback to server as a secrets backend.

**Review by noaleetz — CHANGES_REQUESTED** [review]
last round of changes! I will also ask @andrewtruong for a review!

**Comment by ngrayluna** [comment]
Preview here: 


Screen Shot 2023-11-01 at 6 16 17 PM

**Review by noaleetz — APPROVED** [review]
lgtm!!! Lets just got a review from @gls4 @andrewtruong and we should be good to go :). 

Note that AWS integration will be coming soon - so in a few weeks might have to update

**Review by andrewtruong — APPROVED** [review]

**Comment by noaleetz** [comment]
Got sign-offs from security: @rilwansogbesan and @wandb-kc - [thread](https://weightsandbiases.slack.com/archives/C05754EPT26/p1698948110903129?thread_ts=1696275892.199469&cid=C05754EPT26)

**Review by wandb-kc — APPROVED** [review]

**Review by rilwansogbesan — APPROVED** [review]

**Pull request merged**